synapse/storage/_base.py

```diff
@@ -537,12 +603,45 @@ def _simple_upsert(self, table, keyvalues, values,
 
                 # presumably we raced with another transaction: let's retry.
                 logger.warn(
-                    "IntegrityError when upserting into %s; retrying: %s",
-                    table, e
+                    "%s when upserting into %s; retrying: %s", e.__name__, table, e
```

Comment:
Why this change?

Reply:
Oh, this was from my attempt at making a strategy similar to https://github.com/matrix-org/synapse/pull/4306#issuecomment-450564229 work. It didn't. :(

Reply:
Ah, fair